thegame/engine/game_objects/game_object.py

```diff
@@ -1,43 +1,81 @@
 class GameObject:
     """ An object that will be used to fill the objects sheet of a game."""
 
-    def __init__(self, sprite_location: str, animation: str = None, name: str = None):
-
-        self.sprite_location = sprite_location
+    def __init__(
+        self,
+        sprite_locations: dict,
+        initial_sprite: str = None,
+        animation: str = None,
+        name: str = None,
+    ):
+        """ 
+            sprite_locations {name,location}
+            _loaded_sprites {location, Sprite}
+        """
+        self.sprite_locations = sprite_locations
+        self._loaded_sprites = {}
         self.animation = animation
-        self._loaded_sprite = None
         self.name = name
+        if initial_sprite is None and bool(sprite_locations):
+            self.active_sprite = list(self.sprite_locations.keys())[0]
```

Comment:
I actually wrote that assuming it would pick effectively a random sprite, and not the first inserted sprite since the user didn't specify, but I guess it went that way anyways. It saved on refactor work in the unit tests. I can change it to another option though: 

- set it to None and let the user set the sprite at some point

- make an OrderedDict and use the first inserted sprite

- something else more specific

@Ross-Alexandra

Reply:
After much deliberation, I think that not specifying a sprite should in fact randomly select a sprite. Two thoughts on that though,

1) Explicit code should be written to select a random sprite; right now its different based on the implementation of the version of Python that the user is using.
2) This change should be thoroughly documented, as this is verging on a side effect.

Reply:
Okay will do.

Reply:
Done.